src/rules.ts

```diff
@@ -27,22 +37,28 @@ function isType(v: Primitive, t: typeName): boolean {
     case 'number':
       return isNumber(v);
     case 'boolean':
-      return typeof v == 'boolean';
+      return isBool(v);
+    case 'array':
+      return isArray(v);
   }
 }
 
 const EMPTY_STRING_REGEXP = /^\s*$/;
 
-function isBlank(v: Primitive): boolean {
+function isBlank(v: Value): boolean {
   if (!isDefined(v)) {
     return true;
   }
   if (isString(v)) {
     return EMPTY_STRING_REGEXP.test(v as string);
   }
-  if (isNumber(v) || typeof v === 'boolean') {
+  if (isNumber(v) || isBool(v)) {
     return false;
   }
+  if (isArray(v)) {
+    return v.length === 0;
+  }
+
```

Comment:
@dim we also need to check objects, we already have/will have more things like
`treeId: [v => (v && !!v.id) || 'can't be blank'],`